### PR TITLE
EMP-343: Refactored CRM File Service to be more testable and increase coverage

### PR DESCRIPTION
**Pull request by tanovellino**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-XXX)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java [hunk 1]
```diff
@@ -42,59 +42,41 @@ private static Class<? extends CrmFormModelInterface> getCrmFormTypeMapClass(Int
...
+                crmFormDetailsCriteria.getSpecification(crmFormDetailsCriteriaDTO)
             )
             .orElseThrow(() -> new ResourceNotFoundException("Task with USN " + crmFormDetailsCriteriaDTO.usn() + " not found"));
 
-
         // Collect and clean content
-        String odfImageContentString = new String(task.getCrmFile(), StandardCharsets.UTF_8)
-                .replaceAll("\u0000", "");
+        String crmFormFileContent = new String(task.getCrmFile(), StandardCharsets.UTF_8)
+            .replaceAll("\u0000", "");
```
Comment (gpjustice) [inline equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:62]: Can you put `"\u0000"`  in a string variable - private static final string  ?

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java [hunk 2]
```diff
@@ -109,9 +91,33 @@ public <T extends CrmFormModelInterface> T convertCrmFormJsonToModel(JSONObject
...
+    public JSONObject convertCrmFormLinkedAttachments(JSONObject crmFileJsonObject) {
+        JSONObject linkedAttachments;
+
+        if (!crmFileJsonObject.has("linkedAttachments")) {
+            linkedAttachments = new JSONObject();
+            linkedAttachments.put("linkedAttachment", new JSONArray());
+            return linkedAttachments;
+        }
+
+        linkedAttachments = (JSONObject) crmFileJsonObject.get("linkedAttachments");
```
Comment (gpjustice) [inline equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:103]: same as above

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java [hunk 3]
```diff
@@ -109,9 +91,33 @@ public <T extends CrmFormModelInterface> T convertCrmFormJsonToModel(JSONObject
...
+
+        if (!linkedAttachments.has("linkedAttachment")) {
```
Comment (gpjustice) [inline equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:105]: `linkedAttachment` - same as above, reuse the string variable

File: equinity-historical-data/src/test/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileServiceTest.java [hunk 4]
```diff
@@ -38,25 +41,66 @@ class CrmFileServiceTest {
...
+        crmFileTestTypes.put(5001912L, CRM_TYPE_4);
+        crmFileTestTypes.put(5001604L, CRM_TYPE_5);
+        crmFileTestTypes.put(11L, CRM_TYPE_5);
+        crmFileTestTypes.put(5001662L, CRM_TYPE_7);
+        crmFileTestTypes.put(4808706L, CRM_TYPE_7);
+        crmFileTestTypes.put(5001597L, CRM_TYPE_7);
+
+
+        crmFileTests =  new HashMap<>();
+        crmFileTests.put(5001604L, "src/test/resources/Crm5MockFile_5001604.txt"); // Mocking good XML
```
Comment (gpjustice) [inline equinity-historical-data/src/test/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileServiceTest.java:61]: `src/test/resources` - Do you we need to give the full path?

Comment (tanovellino) [reply equinity-historical-data/src/test/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileServiceTest.java:61]: I think using full path makes it clear which file it refers. Besides,  if it were looking for relative location,  then it wouldn't find the files

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java (hunk 2 above)
Comment (tanovellino) [reply equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:103]: Will create the constant,  just not sure how to name it

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java (hunk 1 above)
Comment (tanovellino) [reply equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:62]: done

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java [hunk 5]
```diff
@@ -109,9 +91,33 @@ public <T extends CrmFormModelInterface> T convertCrmFormJsonToModel(JSONObject
         return (T) crmFormData;
     }
 
-    // TODO (): Delete this, once we have all CRM Form Files available for data-mapping
-    public JSONObject getCrmFormJson(CrmFormDetailsCriteriaDTO crmFormDetailsCriteriaDTO) {
-        return getCrmFileJson(crmFormDetailsCriteriaDTO);
+    public JSONObject convertCrmFormLinkedAttachments(JSONObject crmFileJsonObject) {
+        JSONObject linkedAttachments;
+
+        if (!crmFileJsonObject.has("linkedAttachments")) {
```
[the remark this answers is not in the document]
Comment (tanovellino) [reply equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:97]: done

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java (hunk 2 above)
Comment (tanovellino) [reply equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:103]: done

File: equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java (hunk 3 above)
Comment (tanovellino) [reply equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/service/CrmFileService.java:105]: done

**Review by solomonmanik — APPROVED** [review]

**Pull request merged**